src/main/java/com/elixr/poc/service/UserService.java

```diff
@@ -26,28 +29,32 @@ public UserService(UserRepository userRepository) {
 
     /**
      * Deleting the user by the userId.
-     * throwing a NoRecordFoundException to handel if the UserId is not present.
+     * throwing a GlobalException to handel if the UserId is not present.
      */
-    public boolean deleteUserDetails(UUID userId) throws NoRecordFoundException {
+    public boolean deleteUserDetails(UUID userId) throws IdNotFoundException {
         boolean success = false;
         boolean userRecordExists = userRepository.existsById(userId);
         if (userRecordExists) {
             userRepository.deleteById(userId);
             success = true;
```

Comment:
I don't see success variable used anywhere in the method after you initialized to true . This method is directly returning true after the else part and so you can remove this success variable.

Reply:
removed the unusual declaration of variable success